src/main/java/lotto/domain/LottoRank.java

```diff
@@ -27,10 +26,18 @@ public int getMatchCount() {
         return matchCount;
     }
 
-    public static LottoRank getLottoRank(int matchCount) {
-        return Arrays.stream(values())
+    public static LottoRank getLottoRank(int matchCount, boolean matchBonus) {
+        LottoRank resultRank = Arrays.stream(values())
             .filter(lottoRank -> lottoRank.matchCount == matchCount)
             .findFirst()
             .orElseGet(() -> LottoRank.NO_MATCH);
+
+        if (resultRank.matchCount == LottoRank.SECOND.matchCount && matchBonus) {
+            return LottoRank.SECOND;
+        } else if (resultRank.matchCount == LottoRank.THIRD.matchCount) {
+            return LottoRank.THIRD;
+        }
```

Comment:
다음과 같이 처리해볼 수도 있을 것 같습니다.
```suggestion
        if (resultRank == LottoRank.SECOND && !matchBonus) {
            return LottoRank.THIRD;
        }
```